efficientdet/hparams_config.py

```diff
@@ -91,32 +94,45 @@ def override(self, config_dict_or_str):
     if isinstance(config_dict_or_str, str):
       if not config_dict_or_str:
         return
-      elif "=" in config_dict_or_str:
+      elif config_dict_or_str.endswith('.yaml'):
+        config_dict = self.parse_from_yaml(config_dict_or_str)
+      elif '=' in config_dict_or_str:
         config_dict = self.parse_from_str(config_dict_or_str)
       else:
-        config_dict = self.parse_from_module(config_dict_or_str)
+        raise ValueError(
+            'Invalid string {}, must end with .yaml or contains "=".'.format(
+                config_dict_or_str))
     elif isinstance(config_dict_or_str, dict):
       config_dict = config_dict_or_str
     else:
       raise ValueError('Unknown value type: {}'.format(config_dict_or_str))
 
     self._update(config_dict, allow_new_keys=False)
 
-  def parse_from_module(self, module_name: str) -> dict:
-    """
-    Import config from module_name containing key=value pairs.
-    """
+  def parse_from_module(self, module_name: Text) -> Dict[Any, Any]:
+    """Import config from module_name containing key=value pairs."""
     config_dict = {}
     module = __import__(module_name)
 
     for attr in dir(module):
       # skip built-ins and private attributes
-      if not attr.startswith("_"):
+      if not attr.startswith('_'):
         config_dict[attr] = getattr(module, attr)
 
     return config_dict
 
-  def parse_from_str(self, config_str):
+  def parse_from_yaml(self, yaml_file_path: Text) -> Dict[Any, Any]:
+    """Parses a yaml file and returns a dictionary."""
+    with tf.io.gfile.GFile(yaml_file_path, 'r') as f:
+      config_dict = yaml.load(f, Loader=yaml.FullLoader)
```

Comment:
If this library is pyyaml, `yaml.safe_load` is generally preferred.

Reply:
You are right. safe_load seems to be better.